mlir/include/mlir/Target/LLVMIR/ModuleImport.h

```diff
@@ -277,6 +278,12 @@ class ModuleImport {
   void convertParameterAttributes(llvm::CallBase *call, ArrayAttr &argsAttr,
                                   ArrayAttr &resAttr, OpBuilder &builder);
 
+  /// Whether the importer should try to convert all intrinsics to
+  /// llvm.call_intrinsic instead of dialect supported operations.
+  bool useUnregisteredIntrinsicsOnly() const {
```

Comment:
I wonder if we should start moving the entire unregistered intrinsic handling back into module import:

https://github.com/llvm/llvm-project/blob/f120b0d6d2629e226e6fa75974fbd17f46206bca/mlir/lib/Target/LLVMIR/ModuleImport.cpp#L1651C1-L1652C1

I this function we could try to convert with a dialect interface if preferUnregisteredIntrinsics is not set and then fallback to the translation with call_intrinsic? That way we can make a couple of functions internal to the import and the interface does not need to deal with the unregistered intrinsics?

Reply:
Looks like follow up work attempt? (I do volunteer, but I don't want to mix unrelated work here)